webrtc/datachannel.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{WebRtcError, WebRtcResult};
+
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use uuid::Uuid;
+
+lazy_static! {
```

Comment:
I think these ids should just be numbers and individual webrtc implementations should produce and consume them

We shouldn't have a `DataChannelBackend` trait at all, all of this should be internal to the webrtc impl